### PR TITLE
Update CMakeLists to support cmake build for UTMSP

### PR DESCRIPTION
**Pull request by rpashchapur**
Update CMakeLists to support cmake build for UTMSP

Description
-----------
This PR aims to improve the UTMSP library's build process by updating the CMake and Make build configurations for UTMSP support. It involves configuring required flags, linking the needed libraries, and verifying compatibility.

Test Steps
-----------
To enable and build QGC with UTMSP Adapter support, execute these commands:
1. Create a new build directory: 
```
mkdir build
```
 2. Navigate to the build directory: 
 ```
cd build
```
3. Run cmake with UTM Adapter configuration:
```
cmake -DQGC_CONFIG_UTM_ADAPTER=ON ..
```
4. Compile the build: 
```
make -j8
```

Checklist:
----------

- [x] [Review Contribution Guidelines](https://github.com/mavlink/qgroundcontrol/blob/master/.github/CONTRIBUTING.md).
- [x] [Review Code of Conduct](https://github.com/mavlink/qgroundcontrol/blob/master/.github/CODE_OF_CONDUCT.md).
- [x] I have tested my changes.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/UTMSP/UTMSPAuthorization.cpp [hunk 1]
```diff
@@ -7,66 +7,87 @@
...
+                                                    "0123456789+/";
+
+std::string UTMSPAuthorization::base64_encode(unsigned char const* bytes_to_encode,
+                                              unsigned int in_len)
 {
+    std::string ret;
+    int i = 0, j = 0;
+    unsigned char char_array_3[3], char_array_4[4];
+
+    while (in_len--) {
```
Comment (govindsi) [inline src/UTMSP/UTMSPAuthorization.cpp]: This function  needs change, use QByteArray  helpers for base64 encode

File: src/UTMSP/CMakeLists.txt [hunk 2]
```diff
@@ -1,41 +1,70 @@
...
+
+    target_include_directories(UTMSP
+        PUBLIC
+        services
+    )
+
+    target_link_libraries(UTMSP
+
+        PRIVATE
+        libevents_parser
```
Comment (govindsi) [inline src/UTMSP/CMakeLists.txt]: libevent is available in qt https://doc.qt.io/qt-6/qtwebengine-3rdparty-libevent.html, lets explore if we can migrate to get rid of submodule deps

File: src/UTMSP/UTMSPAuthorization.cpp (hunk 1 above)
Comment (rpashchapur) [reply src/UTMSP/UTMSPAuthorization.cpp]: Remove and replaced it with QByteArray

File: src/UTMSP/CMakeLists.txt (hunk 2 above)
Comment (rpashchapur) [reply src/UTMSP/CMakeLists.txt]: Removed

**Pull request closed without merging**